engine/CMakeLists.txt

```diff
@@ -32,6 +32,17 @@
 
 MESSAGE("PROCESSING DIRECTORY: ${CMAKE_CURRENT_SOURCE_DIR}")
 
+CMAKE_MINIMUM_REQUIRED(VERSION 3.21 FATAL_ERROR)
+
+IF (POLICY CMP0087)
+    cmake_policy(SET CMP0087 NEW)
+ENDIF ()
+SET(X_VCPKG_APPLOCAL_DEPS_INSTALL ON)
+
+IF (POLICY CMP0167)
+    CMAKE_POLICY (SET CMP0167 OLD)
+ENDIF (POLICY CMP0167)
```

Comment:
This is all set in the root CMakeLists.txt so shouldn't be needed here.

Reply:
OK. Fixed. What about the `CMAKE_MINIMUM_REQUIRED` clause? Do you consider that handled adequately at the root level as well?

Reply:
Also, what about the `X_VCPKG_APPLOCAL_DEPS_INSTALL` setting? I'm not sure that that translates to subdirectories.

Reply:
> OK. Fixed. What about the `CMAKE_MINIMUM_REQUIRED` clause? Do you consider that handled adequately at the root level as well?

Yes, that only needs to happen once for the entire project. Let's not confuse ourselves.

> Also, what about the `X_VCPKG_APPLOCAL_DEPS_INSTALL` setting? I'm not sure that that translates to subdirectories.

This one might be more executable dependent - that is, it's specific to the Targets that make an executable. That would be appropriate here.

Reply:
For the `X_VCPKG_APPLOCAL_DEPS_INSTALL` please continue conversation in https://github.com/vegastrike/Vega-Strike-Engine-Source/pull/974#discussion_r1912607454.

Per `CMAKE_MINIMUM_REQUIRED` - let's continue that here

Reply:
OK, I've removed the `CMAKE_MINIMUM_REQUIRED` clauses from everything except the root CMakeLists.txt.